Il2CppInspector.Common/Outputs/IDAPythonScript.cs

```diff
@@ -98,78 +105,129 @@ public class IDAPythonScript
             }
         }
 
-        private void writeMethods(string typeName, IEnumerable<MethodBase> methods) {
+        private void writeMethods(IEnumerable<MethodBase> methods) {
             foreach (var method in methods.Where(m => m.VirtualAddress.HasValue)) {
+                typeGenerator.VisitMethod(method);
+                writeDecls(typeGenerator.GenerateVisitedTypes());
                 var address = method.VirtualAddress.Value.Start;
-                writeName(address, $"{typeName}_{method.Name}");
+                writeTypedName(address, typeGenerator.GenerateMethodDeclaration(method), typeGenerator.MethodNamer.GetName(method));
                 writeComment(address, method);
             }
         }
 
+        private static string sanitizeIdentifier(string str) {
+            return Regex.Replace(str, "[^a-zA-Z0-9_]", "_");
+        }
```

Comment:
It's a bit specific to C output because it's supposed to generate C identifiers (and I'll tweak it to also ensure the name starts with `_` or a letter). So maybe, `String.AsCIdentifier`?

Reply:
That works!